Throttle event backfills

yaoandrew opened the pull request.

yaoandrew committed:
Adds throttler with some duplication

yaoandrew committed:
Moves throttler to utils dir

yaoandrew committed:
Refactors throttler to reduce repetition

ecmendenhall commented:
Just noticed that this PR targets `prod`. There's now a `develop` branch here that deploys to the QA environment, just like in the other repos—let's test this out there before merging to the `prod` branch. The new workflow should be:

- Branch new features off of `develop`
- Merge feature branches to `develop` when done, and they'll deploy to QA.
- If QA looks good, merge `develop` into `prod` to deploy to production.

I'm not sure we've actually exercised the `develop`-to-QA-environment deploy pipeline yet, so there might be a bump or two on the first merge.

ecmendenhall reviewed (changes requested):
Change the target branch to `develop`, otherwise looks good.

yaoandrew commented:
> Change the target branch to `develop`, otherwise looks good.

Sounds good. Changed the base branch and rebased `develop`. I also ran the backfill locally with the new mcdTransformer set and made sure we're picking all the new `dog` and `clip` events.
